optuna/pruners/percentile.py

```diff
@@ -1,14 +1,23 @@
 import math
+import numpy as np
 
 from optuna.pruners import BasePruner
-from optuna.structs import StudyDirection
-from optuna.structs import TrialState
+from optuna import structs
 from optuna import types
 
 if types.TYPE_CHECKING:
     from optuna.storages import BaseStorage  # NOQA
 
 
+def get_best_intermediate_result_over_steps(trial, direction):
```

Comment:
Please add `_` prefix to this function name.

**[FYI]** Prepending `_` to library private functions is a recently added Optuna's convention.